### PR TITLE
Initial project structure

### PR DESCRIPTION
**Pull request by Zanfer322**
Please comment on folder structure.

Also we should use the `milestone` feature of github

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Lint
+
+on: [push, pull_request]
```
Comment (sak96) [inline .github/workflows/lint.yml]: make it only pull request ?

Comment (sak96) [inline .github/workflows/lint.yml]: do we need to add push as well ?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      with:
+        python-version: 3.8
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+
+    - name: black
+      run: black --check ghost_keep tests
```
Comment (sak96) [inline .github/workflows/lint.yml]: why are we running black on test ?

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,27 @@
+__pycache__/
```
Comment (sak96) [inline .gitignore]: add .pyc and other format of file as well ... as well

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+lint:
+	black ghost_keep tests
+	isort -rc ghost_keep tests
+	flake8 ghost_keep
+	mypy ghost_keep
+
+test:
+	pytest tests --cov=ghost_keep
+
+.PHONY: all lint test
```
Comment (sak96) [inline Makefile]: add a new line at the end

File: ghost_keep/first_file.py [hunk 5]
```diff
@@ -0,0 +1,2 @@
+def add_numbers(a: int, b: int) -> int:
```
Comment (sak96) [inline ghost_keep/first_file.py]: no doc string ??
do we want to enforce it ?

File: .github/workflows/lint.yml (hunk 2 above)
Comment (Zanfer322) [reply .github/workflows/lint.yml]: Why give tests special treatment for black. Enforcing black and isort does not take up any extra effort.

File: .gitignore (hunk 3 above)
Comment (Zanfer322) [reply .gitignore]: I have never seen .pyc files so I did not include it, are you sure you want me to include them?

File: ghost_keep/first_file.py (hunk 5 above)
Comment (Zanfer322) [reply ghost_keep/first_file.py]: LOL, this was just for testing. Do you want to enforce docstrings for public functions?

I do not like enforcing docstrings for private methods though

File: Makefile (hunk 4 above)
Comment (Zanfer322) [reply Makefile]: We should be able to add this and trailing whitespace to linting checks, i'll check. Any ideas?

File: .github/workflows/lint.yml (hunk 1 above)
Comment (Zanfer322) [reply .github/workflows/lint.yml]: It is not required, I can remove it

File: .github/workflows/lint.yml (hunk 2 above)
Comment (sak96) [reply .github/workflows/lint.yml]: ok

File: ghost_keep/first_file.py (hunk 5 above)
Comment (sak96) [reply ghost_keep/first_file.py]: is this a private method ?
we should enforce doc strings for public method at least. 
if it private method then let it go to hell.

**Comment by sak96** [comment]
@Zanfer322 
let us follow the `issue -> pr` mechanism where we can discuss things
branch name will be issue_no_description just like in runner

File: .gitignore (hunk 3 above)
Comment (sak96) [reply .gitignore]: in most case it should be in pycache folder 
but including them in git ignore should not be harmful as we are not really need to commit it
`**/*.py[co]`

File: .github/workflows/lint.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
+name: Lint
```
Comment (sak96) [inline .github/workflows/lint.yml]: add wip thing
https://medium.com/dropseed/wip-pull-requests-on-github-b1b713154a46

File: ghost_keep/first_file.py [hunk 7]
```diff
@@ -0,0 +1,2 @@
...
+    return a + b
```
Comment (sak96) [inline ghost_keep/first_file.py]: add docstring to the emtpy files also __init__.py can be used to tell what a folder is for

File: tests/test_add.py [hunk 8]
```diff
@@ -0,0 +1,5 @@
+from ghost_keep.first_file import add_numbers
```
Comment (sak96) [inline tests/test_add.py]: why have you added test file 
two files above this

File: .github/workflows/lint.yml [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+on: [pull_request]
```
[the remark this answers is not in the document]
Comment (sak96) [reply .github/workflows/lint.yml]: https://github.com/marketplace/wip

**Pull request merged**